src/main/java/hudson/tasks/junit/JUnitResultArchiver.java

```diff
@@ -155,6 +174,7 @@ public void perform(Run build, FilePath workspace, Launcher launcher,
 
         if (action != null && action.getResult().getFailCount() > 0)
             build.setResult(Result.UNSTABLE);
+            listener.getLogger().println(Messages.JUnitResultArchiver_ChangeState("UNSTABLE"));
```

Comment:
I believe the conditional above will need braces otherwise this is considered outside of the block and will always execute, which may not have been your intent due to the indentation. If you did want it to always execute, you should update the indentation to reflect that.

Reply:
@aabrahamian Thank you. Corrected it.